File: 004/004.brute.str.py
# brute force solution
# palindrome check via string reverse

def largest_palindrome_product(a: int, b: int) -> int:
  """ Getting the largest palindrome product for two integers.
    Args:
      a (int): The number of digits in the first number.
      b (int): The number of digits in the second number.
    Returns:
      int: the largest palindrome product.
  """
  largest = 0
  for num1 in range(10 ** (a - 1), 10 ** a):
    for num2 in range(10 ** (b - 1), 10 ** b):
      product = num1 * num2
      if product == int(str(product)[::-1]):
        largest = max(largest, product)
  return largest

# results
print(largest_palindrome_product(2, 2))
print(largest_palindrome_product(3, 3))

File: deprecated/004/js/004.brute.str.js
// brute force solution
// palindrome check via string reverse

function largestPalindromeProduct(a, b) {
  /**
   * Searches for the largest palindrome product from two integers.
   * @param {int} a - the number of digits in the first number.
   * @param {int} b - the number of digits in the second number.
   * @returns {int} - the largest found palindrome product.
   */
  let largest = 0;
  for (let num1 = 10 ** (a - 1); num1 < 10 ** a; num1++) {
    for (let num2 = 10 ** (b - 1); num2 < 10 ** b; num2++) {
      let product = num1 * num2;
      if (product == product.toString().split("").reverse().join("")) {
        largest = Math.max(largest, product);
      }
    }
  }
  return largest;
}

// results
console.log(largestPalindromeProduct(2, 2));
console.log(largestPalindromeProduct(3, 3));